taskiq_nats/result_backend.py

```diff
@@ -0,0 +1,122 @@
+        **connect_options: Any,
+    ) -> None:
+        """Construct new result backend.
+
+        :param servers: NATS servers .
+        :param keep_results: flag to not remove results from Redis after reading.
+        :param connect_kwargs: additional arguments for nats `connect()` method.
+        """
+        self.servers: Final = servers
+        self.keep_results: Final = keep_results
+        self.bucket_name: Final = bucket_name
+        self.connect_options: Final = connect_options
+
+        self.nats_client: NATS
+        self.nats_jetstream: JetStreamContext
+        self.object_store: ObjectStore
+
+    async def startup(self) -> None:
+        """Create new connection to NATS.
+
+        Initialize JetStream context and new ObjectStore instance.
+        """
+        self.nats_client = await nats.connect(
+            servers=self.servers,
+            **self.connect_options,
+        )
+        self.nats_jetstream = self.nats_client.jetstream()
+
+        try:
+            self.object_store = await self.nats_jetstream.object_store(self.bucket_name)
+        except BucketNotFoundError:
+            self.object_store = await self.nats_jetstream.create_object_store(
+                self.bucket_name,
+            )
+
+    async def shutdown(self) -> None:
+        """Close nats connection."""
+        if self.nats_client.is_closed:
+            return
+        await self.nats_client.close()
+
+    async def set_result(self, task_id: str, result: TaskiqResult[_ReturnType]) -> None:
+        """Set result to the nats bucket.
+
+        :param task_id: ID of the task.
+        :param result: result of the task.
+        """
+        await self.object_store.put(
+            name=task_id,
+            data=pickle.dumps(result),
+        )
+
+    async def is_result_ready(self, task_id: str) -> bool:
+        """Returns whether the result is ready.
+
+        :param task_id: ID of the task.
+
+        :returns: True if the result is ready else False.
+        """
+        try:
+            await self.object_store.get(name=task_id)
+        except ObjectNotFoundError:
+            return False
+        return True
+
+    async def get_result(
+        self,
+        task_id: str,
+        with_logs: bool = False,
+    ) -> TaskiqResult[_ReturnType]:
+        """
+        Retrieve result from the task.
+
+        :param task_id: task's id.
+        :param with_logs: if True it will download task's logs.
+        :raises ResultIsMissingError: if there is no result when trying to get it.
+        :return: TaskiqResult.
+        """
+        try:
+            result = await self.object_store.get(
+                name=task_id,
+            )
+        except ObjectNotFoundError as exc:
+            raise ResultIsMissingError from exc
+
+        if not self.keep_results:
+            await self.object_store.delete(
+                name=task_id,
+            )
+
+        taskiq_result: TaskiqResult[_ReturnType] = pickle.loads(  # noqa: S301
```

Comment:
Can you replace pickle with TaskiqSerializers as [here](https://github.com/taskiq-python/taskiq-redis/blob/65a0b6c431e8cb256817f4ab6455d9ab043078f6/taskiq_redis/redis_backend.py#L120).